li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -300,6 +347,8 @@ synchronized public void subscribe(Pattern pattern) {
 
   @Override
   synchronized public void unsubscribe() {
+    ensureOpen();
```

Comment:
this is public, yet does not close/clear existing consumers (or pass the unsub() to them) - is this a potential bug if this is called externally ?

Reply:
My bad. I didn't provide the full implementation of this method here, because I planned to do that when I add subscribe() implementation). And I added these lines just not to forget to change the subscription state.